### ISSUE TITLE
Customizable binary name & install path instructions 

### ISSUE THREAD

**Issue by naliferopoulos** [Airblader/i3#254]
## I'm submitting a…

<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Currently, the way i3-gaps installation instructions are laid out, the i3-gaps binary that is built replaces the original i3 binary. That might be newbie friendly in the sense that much less configuration will be needed to external systems when migrating from i3 to i3-gaps, however this creates an issue when the original i3 package is installed by a package management system.

## Desired Behavior

I would appreciate you provided build instructions to customize the installation paths and binary name so that conflicts between official i3 packages and i3-gaps can at least be manually resolved.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 4.16-37-gdbab1538 (2018-11-10, branch "gaps-next") © 2009 Michael Stapelberg and contributors
</pre>

<pre>
- Linux Distribution & Version: 4.18.0-kali2-amd64
- Are you using a compositor (e.g., xcompmgr or compton): Compton is in use.
</pre>

**Comment by Airblader** [comment]
Hi, while I understand where you're coming from, there's a few different issues and considerations for this.

1) The build instructions are mostly for package maintainers, not for end users (and the wiki says that). I would assume that a package maintainer knows how to rename a binary if that's what they want to do. I do not intend for users to build from source and do not support them in doing so.

2) i3(-gaps) doesn't ship a binary, but around a dozen or so. Since it's pretty unclean to replace one and not others, we'd have to rename them all. This would break a lot of things: binaries called from within i3 itself (i3bar, i3-nagbar) as well as all kinds of scripts and tools out there (using things like i3, i3-msg, …).

3) i3-gaps is a superset of i3, sort to speak. If you don't enable any of the i3-gaps features, you have a completely normally functioning i3. Thus there's really no reason to want to install both side-to-side. If you need an i3-gaps feature, install i3-gaps. Otherwise, install i3.

> however this creates an issue when the original i3 package is installed by a package management system.

Assuming the packages are set up in a reasonable way, there should be no such issue. For example on Arch, there's things like "groups" and "provides". The i3 and i3-gaps packages can coexist without any problem, and installing either one replaces the other (for which you will be warned and asked for confirmation).

To summarize: renaming breaks things, there's no real need to do it and if you do have a reason to do it, it's a packaging concern first and foremost and can thus be handled in the package definition. Thanks for understanding!
